File: 프로그래머스/1/42748. K번째수/K번째수.swift
import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    
    var result: [Int] = []
    
    for cmd in commands { // 50번
        var arr = array[cmd[0]-1...cmd[1]-1].sorted() // (100) * 100lg100
        result.append(arr[cmd[2]-1])    
    }
    
    return result
}

// array: 1~100이하
// 원소 1~100이하
// commands: 1~50이하

// cmd[0] ~ cmd[1]까지 배열 슬라이싱
// sort
// k번째 요소 result에 주입

File: 프로그래머스/1/42748. K번째수/K번째수.py
# 자르고 정렬했을때 K번쨰 수
# 100개 배열, 50번의 연산
# 

def solution(array, commands):
    answer = []
    for i, j, k in commands:
        part = sorted(array[i-1:j])
        answer.append(part[k-1])
    return answer